source/type/message.h

```diff
@@ -263,6 +263,26 @@ struct AuxInfoForMetaVJ {
 };
 }  // namespace detail
 
+struct TimeSlot {
+    uint32_t begin = 0;  ///< seconds since midnight
+    uint32_t end = 0;    ///< seconds since midnight
+    TimeSlot(uint32_t begin = 0, uint32_t end = 0) : begin{begin}, end{end} {}
+
+    template <class Archive>
+    void serialize(Archive& ar, const unsigned int);
+};
+
+struct ApplicationPattern {
+    navitia::type::WeekPattern week_pattern;
+    boost::gregorian::date_period application_period{boost::gregorian::date(), boost::gregorian::date()};
+    std::vector<TimeSlot> time_slots;
+
+    void add_time_slot(uint32_t begin, uint32_t end);
+    template <class Archive>
+    void serialize(Archive& ar, const unsigned int);
+    void sort_time_slots();
```

Comment:
you can use `std::set` instead of `set::vector`

Reply:
:+1: